rclpy/rclpy/parameter_client.py

```diff
@@ -0,0 +1,329 @@
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def get_parameter_types(
+        self,
+        names: List[str],
+        callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Get parameter types given names.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.GetParameterTypes.Response``.
+
+        Parameter type definitions are given in Parameter.Type
+
+        :param names: List of parameter names to get types for.
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = GetParameterTypes.Request()
+        request.names = names
+        future = self._get_parameter_types_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def set_parameters_atomically(
+        self,
+        parameters: Sequence[Union[Parameter, ParameterMsg]],
+        callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Set parameters atomically.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.SetParametersAtomically.Response``.
+
+        :param parameters: Sequence of parameters to set.
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = SetParametersAtomically.Request()
+        request.parameters = [
+            param.to_parameter_msg()
+            if isinstance(param, Parameter) else param
+            for param in parameters
+        ]
+        future = self._set_parameters_atomically_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def delete_parameters(
+        self, names: List[str], callback: Optional[Callable] = None
+    ) -> Future:
+        """
+        Unset parameters with given names.
+
+        The result after the returned future is complete
+        will be of type ``rcl_interfaces.srv.SetParameters.Response``.
+
+        Note: Only parameters that have been declared as dynamically typed can be unset.
+
+        :param names: List of parameter names to unset.
+        :param callback: Callback function to call when the request is complete.
+        :return: ``Future`` with the result of the request.
+        """
+        request = SetParameters.Request()
+        request.parameters = [Parameter(name=i).to_parameter_msg() for i in names]
+        future = self._set_parameter_client.call_async(request)
+        if callback:
+            future.add_done_callback(callback)
+        return future
+
+    def load_parameter_file(
+        self,
+        parameter_file: str,
+        use_wildcard: bool = False,
```

Comment:
I think it would be nice to add a callback argument for the `load_parameter_file` and `load_parameter_file_atomically` methods, considering we are supporting callbacks for all of the other parameter client operations.